### DISCUSSION TITLE
What are the reasons why I should choose ntex over actix-web or axum?

### DISCUSSION THREAD

**Discussion by juddbaguio** [ntex-rs/ntex#289] [Q&A] [answered]
I realized that @fafhrd91 was the one who started actix-web and ntex but is now fully-committed to ntex (correct me if I'm wrong). Since the APIs are somewhat similar for both ntex and actix-web, what are the reasons why I should prefer on using ntex over actix-web? Same question as to why choose ntex over axum?

**Comment by fafhrd91** [comment]
you probably should use axum, it is safer option

**Comment by ConsoleC137** [comment, marked as answer]
If you are writing a web server for yourself, you can use any of these three. At the moment, I think `actix-web` is better for production. It is stable, fast (on average 10% faster than `axum` in my tests), and has integration with `rustls` (important for me).
